ql/src/java/org/apache/hadoop/hive/ql/QueryCompilationBreakdownSummary.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.hive.ql.exec.tez.monitoring.PrintSummary;
+import org.apache.hadoop.hive.ql.log.PerfLogger;
+import org.apache.hadoop.hive.ql.session.SessionState;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.hadoop.hive.ql.exec.tez.monitoring.Constants.SEPARATOR;
+
+public class QueryCompilationBreakdownSummary implements PrintSummary {
+  private static final String OPERATION_SUMMARY = "%-84s %9s";
+  private static final String QUERY_COMPILATION_SUMMARY = "Query Compilation Summary";
+  private static final String LINE_SEPARATOR = "line.separator";
+  private static final String TOTAL_COMPILATION_TIME = "Total Compilation Time";
+  private List<Pair<String, String>> compileSteps = new ArrayList<>();
+
+  private String format(String value, long number) {
+    return String.format(OPERATION_SUMMARY, value, number + "ms");
+  }
+
+  private void populateCompileStepNumbers() {
+    compileSteps.add(Pair.of(PerfLogger.PARSE, "1"));
+    compileSteps.add(Pair.of(PerfLogger.GENERATE_RESOLVED_PARSETREE, "2"));
+    compileSteps.add(Pair.of(PerfLogger.LOGICALPLAN_AND_HIVE_OPERATOR_TREE, "3"));
+    compileSteps.add(Pair.of(PerfLogger.GENERATE_LOGICAL_PLAN, "3.1"));
+    compileSteps.add(Pair.of(PerfLogger.PLAN_GENERATION, "3.1.1"));
+    compileSteps.add(Pair.of(PerfLogger.MV_REWRITE_FIELD_TRIMMER, "3.1.2"));
```

Comment:
The hardcoded string numbering may be hard to maintain if we decide to add or remove steps in the future since it will require manually updating the counters.

Using a dedicated structure for keeping the steps could alleviate this by making the implicit nesting explicit. For instance:
```java
private static final CompileStep {
    private final String name;
    private final List<CompileStep> substeps;
}
```